File: WorkerNode/src/main/java/com/yahoo/sdvornik/sorter/BubbleSort.java
package com.yahoo.sdvornik.sorter;

/**
 * Implementation of BubbleSort algorithm.
 */
public final class BubbleSort extends Sorter {
    /**
     * Ctor.
     * @param longArr
     */
    public BubbleSort(long[] longArr) {
        super(longArr);
    }

    public long[] sort() {
		for (int i = 0; i < longArr.length; ++i) {
			for (int j = 0; j < longArr.length - i - 1; ++j) {
				if (longArr[j] > longArr[j + 1]) swap(j,j+1);
	        }
		}
		return longArr;
	}
}

File: worker/src/main/kotlin/com/yahoo/sdvornik/sorter/BubbleSort.kt
package com.yahoo.sdvornik.sorter

/**
 * @author Serg Dvornik <sdvornik@yahoo.com>
 */
class BubbleSort
/**
 * Ctor.
 * @param longArr
 */
(longArr: LongArray) : Sorter(longArr) {

    override fun sort(): LongArray {
        for (i in longArr.indices) {
            for (j in 0..longArr.size - i - 1 - 1) {
                if (longArr[j] > longArr[j + 1]) swap(j, j + 1)
            }
        }
        return longArr
    }
}